### DISCUSSION TITLE
WARden Terminal

### DISCUSSION THREAD

**Discussion by nyxnor** [raspiblitz/raspiblitz#3927] [Feature / App Ideas]
There is the web WebUI version and the terminal version https://github.com/pxsocs/warden_terminal
@pxsocs

Requests to price use tor.

Succesfull install:
![temp](https://user-images.githubusercontent.com/69700936/123519606-eaa6b180-d69b-11eb-992d-36432a787fa2.jpeg)

There is an API key which expired doing my tests:
![temp](https://user-images.githubusercontent.com/69700936/123519570-a9160680-d69b-11eb-84b5-edefa25da6dc.png)

Due to having to deal with key expiration, I'm not sure if I should submit a PR for this.

Here it is the `bonus.warden_terminal.sh`

```bash
#!/bin/bash

# https://github.com/pxsocs/warden_terminal

# command info
if [ $# -eq 0 ] || [ "$1" = "-h" ] || [ "$1" = "-help" ]; then
 echo "config script to install or uninstall WARden Terminal"
 echo "bonus.warden_terminal.sh [on|off|menu]"
 exit 1
fi

source /mnt/hdd/raspiblitz.conf

# add default value to raspi config if needed
if ! grep -Eq "^warden_terminal=" /mnt/hdd/raspiblitz.conf; then
  echo "warden_terminal=off" >> /mnt/hdd/raspiblitz.conf
fi

# show info menu
if [ "$1" = "menu" ]; then
  dialog --title " Info WARden Terminal " --msgbox "
WARden Terminal is a command line tool.
Type: 'wt' in the command line to switch to the dedicated user.
Usage: https://github.com/pxsocs/warden_terminal/blob/master/README.md
" 10 75
  exit 0
fi

# install
if [ "$1" = "1" ] || [ "$1" = "on" ]; then

  echo "*** INSTALL WARDEN TERMINAL ***"

  # create wt user
  sudo adduser --disabled-password --gecos "" wt

  # download source code
  sudo -u wt git clone https://github.com/pxsocs/warden_terminal.git /home/wt/warden_terminal

  sudo -u wt pip3 install -r /home/wt/warden_terminal/requirements.txt
  sudo -u wt python3 /home/wt/warden_terminal/node_warden.py

  # setting value in raspi blitz config
  sudo sed -i "s/^warden_terminal=.*/warden_terminal=on/g" /mnt/hdd/raspiblitz.conf

  echo "# Usage: https://github.com/pxsocs/warden_terminal/blob/master/README.md"
  echo "# To start type: 'sudo su wt' in the command line."
  echo "# To exit the user - type 'exit' and press ENTER"

  exit 0

fi

# switch off
if [ "$1" = "0" ] || [ "$1" = "off" ]; then

  # setting value in raspi blitz config
  sudo sed -i "s/^warden_terminal=.*/warden_terminal=off/g" /mnt/hdd/raspiblitz.conf

  echo "*** REMOVING WARDEN TERMINAL ***"
  sudo userdel -rf wt
  echo "# OK, wt is removed."
  exit 0

fi
```

**Comment by nyxnor** [comment]
PXSocs explained the key provided by default is limited, one should create a free account to request for more keys that will have 100k calls. 250k the key ends and need to create another account that just requires e-mail. https://min-api.cryptocompare.com/pricing

Their license is for personal use, so don't know if you can sell this https://www.cryptocompare.com/free-api-use/

![temp](https://user-images.githubusercontent.com/69700936/123520190-0790b400-d69f-11eb-8194-d73c2dd21e6c.png)

**Comment by pxsocs** [comment]
Thanks for this @nyxnor 
As a separate note, working on an upgrade that will include Raspberry pi specific info like: Temperature, voltage, cpu usage and others.

**Comment by nyxnor** [comment]
https://twitter.com/alphaazeta/status/1419667516922765316/photo/1

**Comment by pxsocs** [comment]
Latest Version has been tested on RaspiBlitz. Features:
. Auto-detects if running at a RaspiBlitz node - [checks for `raspiblitz.info` file](https://github.com/pxsocs/warden_terminal/blob/b827e2a9ff53934ac8fdbc671c0548be6270cc3a/node_warden.py#L482) 
. Outputs to `/dev/tty1` or a different output by editing [`config.ini`](https://github.com/pxsocs/warden_terminal/blob/b827e2a9ff53934ac8fdbc671c0548be6270cc3a/config.ini#L11)
. Automatically detects Bitcoin [RPC setting](https://github.com/pxsocs/warden_terminal/blob/b827e2a9ff53934ac8fdbc671c0548be6270cc3a/rpc.py#L15)
. Shows Sync Status (see below)

Would be great to get additional testers to run and provide feedback. 

https://github.com/pxsocs/warden_terminal

![image](https://user-images.githubusercontent.com/40473443/127744564-eb266e36-506b-42bd-aabc-b638b4c0dfd9.png)

**Comment by nyxnor** [comment]
@pxsocs Working. Liking the auto scrolling and slides type.

The script detects that I am using Raspiblitz, but this variable is not printed correct here: https://github.com/pxsocs/warden_terminal/blob/1ef6727b20885ce747e1521b13cd58ae40611d09/node_warden.py#L477

![Screenshot 2021-08-01 at 14 29 38](https://user-images.githubusercontent.com/69700936/127770877-2b75f26d-9132-474d-973c-4c54b7b8ec69.png)

**Comment by pxsocs** [comment]
Fixed for next version. Missed an `f` string :) 
`logging.info(f"[INFO] Autodetected node: {node}")`

**Comment by rootzoll** [comment]
OK. Sorry for having pushed this so long ahead - the RaspiBlitz WebUI was sucking up all my attention. So whats the status of WARDen RaspiBlitz integration? I dont see a PR for the Terminal Version and the WebUI PR #1678 & #1675 looks unmaintained and still based on RaspiBlitz v1.6 branch. Looking to release RaspiBlitz v1.8.1 end of October ... let me know if it still makes sense add this for the release.
